### PR TITLE
Migrate from DOP to DOT

### PR DESCRIPTION
**Pull request by pwnage101**

### CODE REVIEW

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py [hunk 1]
```diff
@@ -56,7 +57,7 @@ def test_get_uuid(self):
         """ Verify the endpoint returns the details for a single course with UUID. """
         url = reverse('api:v1:course-detail', kwargs={'key': self.course.uuid})
 
-        with self.assertNumQueries(FuzzyInt(45, 3)):
+        with self.assertNumQueries(FuzzyInt(43, 2)):
```
Comment (mikix) [inline course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Is that undoing a recent change? I think we increased the fuzz recently.

File: course_discovery/apps/core/tests/factories.py [hunk 2]
```diff
@@ -54,11 +54,8 @@ class PartnerFactory(factory.DjangoModelFactory):
     marketing_site_api_password = factory.Faker('password')
     analytics_url = factory.Faker('url')
     analytics_token = factory.Faker('sha256')
-    lms_url = FuzzyUrlRoot().fuzz()
+    lms_url = "http://127.0.0.1:8000"
```
Comment (mikix) [inline course_discovery/apps/core/tests/factories.py:57]: Why this?

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py (hunk 1 above)
Comment (pwnage101) [reply course_discovery/apps/api/v1/tests/test_views/test_courses.py]: Hm, so you changed it from `FuzzyInt(46, 2)` to `FuzzyInt(45, 3)`.

Then, meanwhile we changed it from `FuzzyInt(46, 2)` to `FuzzyInt(43, 2)`.

I just selected the latter while resolving a merge conflict.  How does `FuzzyInt(43, 3)` sound as a compromise?

Comment (mikix) [reply course_discovery/apps/api/v1/tests/test_views/test_courses.py]: I'm not picky - just wanted to make sure it wasn't a bad merge. `43, 3` seems fine.

File: course_discovery/apps/core/tests/factories.py (hunk 2 above)
Comment (pwnage101) [reply course_discovery/apps/core/tests/factories.py:57]: No clue, I missed it.  FYI @nasthagiri , I'm going to revert this and see what travis says.

Comment (pwnage101) [reply course_discovery/apps/core/tests/factories.py:57]: Keeping the lms_url fuzzy causes the auth flow to fail in a few api tests.  We established that this seems non-critical, so we're just going to push through with the hard-coded string and address the neatness of this test at the end of hackathon.

**Comment by nasthagiri** [comment]
FYI - I closed and re-opened this PR to re-trigger Travis.

**Comment by nasthagiri** [comment]
I'm merging this PR - so it can be on Stage before the refresh_course_metadata jenkins job runs tomorrow morning at 8:30am.

**Comment by mikix** [comment]
I just got a 500 when hitting discovery.stage.edx.org/login

**Comment by nasthagiri** [comment]
@mikix Thanks for pointing that out.  It is now working.  We needed to add `user_id` as an allowed scope for the new application.  (Needing to add `user_id` scope is a step that we should eliminate soon.)

**Comment by nedbat** [comment]
@pwnage101 An operator has questions/concerns about this PR: https://discuss.openedx.org/t/problems-with-new-changes-in-discovery-documentation/995

Do you mind taking a look?

**Pull request merged**